cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -637,35 +644,69 @@ impl<T: Config> Pallet<T> {
 		});
 	}
 
-	fn enqueue_xcmp_message(
+	fn enqueue_xcmp_messages(
 		sender: ParaId,
-		xcm: BoundedVec<u8, MaxXcmpMessageLenOf<T>>,
+		xcms: &[BoundedVec<u8, MaxXcmpMessageLenOf<T>>],
+		meter: &mut WeightMeter,
 	) -> Result<(), ()> {
 		let QueueConfigData { drop_threshold, .. } = <QueueConfig<T>>::get();
-		let fp = T::XcmpQueue::footprint(sender);
-		// Assume that it will not fit into the current page:
-		let new_pages = fp.ready_pages.saturating_add(1);
-		if new_pages > drop_threshold {
-			// This should not happen since the channel should have been suspended in
-			// [`on_queue_changed`].
-			log::error!("XCMP queue for sibling {:?} is full; dropping messages.", sender);
-			return Err(())
+		let batches_footprints = T::XcmpQueue::get_batches_footprints(
+			sender,
+			xcms.iter().map(|xcm| xcm.as_bounded_slice()),
+			drop_threshold,
+		);
+
+		let best_batch_idx = batches_footprints.binary_search_by(|batch_info| {
```

Comment:
How can we assume that `get_batches_footprints` returned it sorted in the order that we want?  
Can you please add some comments on what is happening?

Reply:
I had the exact same thought/question when I was checking this in the morning (but didn't finish).

Reply:
Done. Added a comment.